base/v0_5_exp/translate_test.go

```diff
@@ -1616,6 +1616,114 @@ func TestTranslateKernelArguments(t *testing.T) {
 	}
 }
 
+// TestTranslateSSHAuthorizedKeyResource tests translating the butane passwd.users[i].ssh_authorized_keys[j] entries to ignition passwd.users[i].ssh_authorized_keys[j] entries.
+func TestTranslateSSHAuthorizedKeyResource(t *testing.T) {
+	tempDir := t.TempDir()
+	var sshKey = "ssh-rsa AAAAAAAAA"
+	var sshKeyEmpty = ""
+	var sshKeyFileName = "id_rsa.pub"
+	var sshKeyOtherFileName = "id_ed25519.pub"
+	var sshKeyEmptyFileName = "empty.pub"
+	err := ioutil.WriteFile(filepath.Join(tempDir, sshKeyFileName), []byte(sshKey), 0644)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	err = ioutil.WriteFile(filepath.Join(tempDir, sshKeyEmptyFileName), []byte(""), 0644)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	tests := []struct {
+		in      SSHAuthorizedKeyResource
+		out     types.SSHAuthorizedKey
+		report  report.Report
+		fileDir string
+	}{
+		{
+			SSHAuthorizedKeyResource{},
+			types.SSHAuthorizedKey(""),
+			report.Report{Entries: []report.Entry{}},
+			tempDir,
+		},
+		{
+			SSHAuthorizedKeyResource{
+				Inline: (*SSHAuthorizedKey)(&sshKey),
+			},
+			types.SSHAuthorizedKey(sshKey),
+			report.Report{Entries: []report.Entry{}},
+			tempDir,
+		},
+		{
+			SSHAuthorizedKeyResource{Local: &sshKeyFileName},
+			types.SSHAuthorizedKey(sshKey),
+			report.Report{Entries: []report.Entry{}},
+			tempDir,
+		},
+		{
+			SSHAuthorizedKeyResource{Inline: (*SSHAuthorizedKey)(&sshKeyEmpty)},
+			types.SSHAuthorizedKey(""),
+			report.Report{Entries: []report.Entry{
+				{
+					Kind:    report.Error,
+					Message: "a message",
+				},
+			}},
+			tempDir,
+		},
+		{
+			SSHAuthorizedKeyResource{Local: &sshKeyOtherFileName},
+			types.SSHAuthorizedKey(""),
+			report.Report{Entries: []report.Entry{
+				{
+					Kind:    report.Error,
+					Message: "a message",
+				},
+			}},
+			tempDir,
+		},
+		{
+			SSHAuthorizedKeyResource{Local: &sshKeyEmptyFileName},
+			types.SSHAuthorizedKey(""),
+			report.Report{Entries: []report.Entry{
+				{
+					Kind:    report.Error,
+					Message: "a message",
+				},
+			}},
+			tempDir,
+		},
+		{
+			SSHAuthorizedKeyResource{Local: &sshKeyFileName},
+			types.SSHAuthorizedKey(""),
+			report.Report{Entries: []report.Entry{
+				{
+					Kind:    report.Error,
+					Message: "a message",
+				},
+			}},
+			"",
+		},
+		{
+			SSHAuthorizedKeyResource{Local: &sshKeyFileName},
+			types.SSHAuthorizedKey(""),
+			report.Report{Entries: []report.Entry{
+				{
+					Kind:    report.Error,
+					Message: "a message",
+				},
+			}},
+			t.TempDir(),
+		},
+	}
+
+	for i, test := range tests {
+		actual, _, r := translateSSHAuthorizedKeyResource(test.in, common.TranslateOptions{FilesDir: test.fileDir})
+		assert.Equal(t, test.out, actual, "#%d: translation mismatch", i)
+		assert.Equal(t, len(test.report.Entries), len(r.Entries), "#%d: report-size mismatch", i)
```

Comment:
Let's check the report contents too.  We usually just compare the value of `r.String()`.